# **Ejercicios de pair programming Módulo 3 Sprint 2**
## **Tableau**
### **Limpieza del dataframe**

---

In [3]:
import pandas as pd
import numpy as nd
import seaborn as sns
import pandas as pd
import matplotlib.pyplot as plt

In [4]:
cuernos = pd.read_csv("../datos/affairs.csv", index_col = 0)

In [7]:
def df_explore(dataframe):
    print(
        f"El dataframe tiene {dataframe.shape[0]} filas y {dataframe.shape[1]} columnas")
    print("--------------")
    print(f"El dataframe tiene las siguientes columnas:")
    for col in dataframe.columns:
        print(col)
    print("--------------")
    print(f"Los principales estadísticos de las variables numéricas son:")
    display(dataframe.describe().T)
    print("--------------")
    print(f"Los principales estadísticos de las variables categóricas son:")
    display(dataframe.describe(include='object').T)
    print("--------------")
    print("El porcentaje de nulos por columna:")
    for i, col in enumerate(dataframe.isnull().sum()):
        print(
            f"{dataframe.isnull().sum().index[i]}: {col/dataframe.shape[0]*100}")
    print("--------------")
    print(f"Hay {dataframe.duplicated().sum()} filas duplicadas.")
    print("--------------")
    print("A continuación el resultado del método .info():")
    dataframe.info()

In [8]:
df_explore(cuernos)

El dataframe tiene 601 filas y 9 columnas
--------------
El dataframe tiene las siguientes columnas:
affairs
gender
age
yearsmarried
children
religiousness
education
occupation
rating
--------------
Los principales estadísticos de las variables numéricas son:


count       mean       std     min   25%   50%   75%   max
affairs        601.0   1.455907  3.298758   0.000   0.0   0.0   0.0  12.0
age            601.0  32.487521  9.288762  17.500  27.0  32.0  37.0  57.0
yearsmarried   601.0   8.177696  5.571303   0.125   4.0   7.0  15.0  15.0
religiousness  601.0   3.116473  1.167509   1.000   2.0   3.0   4.0   5.0
education      601.0  16.166389  2.402555   9.000  14.0  16.0  18.0  20.0
occupation     601.0   4.194676  1.819443   1.000   3.0   5.0   6.0   7.0
rating         601.0   3.931780  1.103179   1.000   3.0   4.0   5.0   5.0

--------------
Los principales estadísticos de las variables categóricas son:


count unique     top freq
gender     601      2  female  315
children   601      2     yes  430

--------------
El porcentaje de nulos por columna:
affairs: 0.0
gender: 0.0
age: 0.0
yearsmarried: 0.0
children: 0.0
religiousness: 0.0
education: 0.0
occupation: 0.0
rating: 0.0
--------------
Hay 24 filas duplicadas.
--------------
A continuación el resultado del método .info():
<class 'pandas.core.frame.DataFrame'>
Int64Index: 601 entries, 4 to 9010
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   affairs        601 non-null    int64  
 1   gender         601 non-null    object 
 2   age            601 non-null    float64
 3   yearsmarried   601 non-null    float64
 4   children       601 non-null    object 
 5   religiousness  601 non-null    int64  
 6   education      601 non-null    int64  
 7   occupation     601 non-null    int64  
 8   rating         601 non-null    int64  
dtypes: float64(2), int64(5), object(2)
memory usage: 47.0+ KB


In [9]:
cuernos.head()

affairs  gender   age  yearsmarried children  religiousness  education  \
4         0    male  37.0         10.00       no              3         18   
5         0  female  27.0          4.00       no              4         14   
11        0  female  32.0         15.00      yes              1         12   
16        0    male  57.0         15.00      yes              5         18   
23        0    male  22.0          0.75       no              2         17   

    occupation  rating  
4            7       4  
5            6       4  
11           1       4  
16           6       5  
23           6       3

In [11]:
cuernos.reset_index(drop = True)

affairs  gender   age  yearsmarried children  religiousness  education  \
0          0    male  37.0         10.00       no              3         18   
1          0  female  27.0          4.00       no              4         14   
2          0  female  32.0         15.00      yes              1         12   
3          0    male  57.0         15.00      yes              5         18   
4          0    male  22.0          0.75       no              2         17   
..       ...     ...   ...           ...      ...            ...        ...   
596        1    male  22.0          1.50      yes              1         12   
597        7  female  32.0         10.00      yes              2         18   
598        2    male  32.0         10.00      yes              2         17   
599        2    male  22.0          7.00      yes              3         18   
600        1  female  32.0         15.00      yes              3         14   

     occupation  rating  
0             7       4  
1             6       4  
2             1       4  
3             6       5  
4             6       3  
..          ...     ...  
596           2       5  
597           5       4  
598           6       5  
599           6       2  
600           1       5  

[601 rows x 9 columns]

**Description**

Infidelity data, known as Fair's Affairs. Cross-section data from a survey conducted by Psychology Today in 1969.

**Format**

A data frame containing 601 observations on 9 variables.

**affairs**

numeric. How often engaged in extramarital sexual intercourse during the past year? 0 = none, 1 = once, 2 = twice, 3 = 3 times, 7 = 4–10 times, 12 = monthly, 12 = weekly, 12 = daily.

**gender**

factor indicating gender.

**age**

numeric variable coding age in years: 17.5 = under 20, 22 = 20–24, 27 = 25–29, 32 = 30–34, 37 = 35–39, 42 = 40–44, 47 = 45–49, 52 = 50–54, 57 = 55 or over.

**yearsmarried**

numeric variable coding number of years married: 0.125 = 3 months or less, 0.417 = 4–6 months, 0.75 = 6 months–1 year, 1.5 = 1–2 years, 4 = 3–5 years, 7 = 6–8 years, 10 = 9–11 years, 15 = 12 or more years.

**children**

factor. Are there children in the marriage?

**religiousness**

numeric variable coding religiousness: 1 = anti, 2 = not at all, 3 = slightly, 4 = somewhat, 5 = very.

**education**

numeric variable coding level of education: 9 = grade school, 12 = high school graduate, 14 = some college, 16 = college graduate, 17 = some graduate work, 18 = master's degree, 20 = Ph.D., M.D., or other advanced degree.

**occupation**

numeric variable coding occupation according to Hollingshead classification (reverse numbering).

**rating**

numeric variable coding self rating of marriage: 1 = very unhappy, 2 = somewhat unhappy, 3 = average, 4 = happier than average, 5 = very happy.

In [12]:
diccionario_age = {17.5 : 'under 20', 22 : '20–24', 27 : '25–29', 32 : '30–34', 37 : '35–39', 42 : '40–44', 47 : '45–49', 52 : '50–54', 57 : '55 or over'}

In [ ]:
diccionario_affaris = {0 : 'none', 1 : 'once', 2 : 'twice', 3 : '3 times', 7 : '4–10 times', 12 : 'monthly', 12 : 'weekly', 12 : 'daily'}

diccionario_yearsmarried = {0.125 : '3 months or less', 0.417 : '4–6 months', 0.75 : '6 months–1 year', 1.5 : '1–2 years', 4 : '3–5 years', 7 : '6–8 years', 10 : '9–11 years', 15 : '12 or more years'}

diccionario_religiousness = {1 : 'anti', 2 : 'not at all', 3 : 'slightly', 4 : 'somewhat', 5 : 'very'}

diccionario_education = {9 : 'grade school', 12 : 'high school graduate', 14 : 'some college', 16 : 'college graduate', 17 : 'some graduate work', 18 : "master's degree", 20 : "Ph.D., M.D., or other advanced degree"}

diccionario_rating = {1 : 'very unhappy', 2 : 'somewhat unhappy', 3 : 'average', 4 : 'happier than average', 5 : 'very happy'}

